.github/workflows/ci.yml

```diff
@@ -3,71 +3,292 @@ name: Build/Test/Release
+
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.toolchain || 'stable' }}
+          target: ${{ matrix.target }}
+          components: ${{ matrix.components || 'rustfmt, clippy' }}
+
+      - name: Cargo fmt
+        uses: clechasseur/rs-cargo@v2
+        with:
+          use-cross: ${{ matrix.cross }}
+          command: fmt
+          args: --all -- --check
+
+      - name: Cargo clippy
+        uses: clechasseur/rs-cargo@v2
+        with:
+          use-cross: ${{ matrix.cross }}
+          command: clippy
+          args: --all --target ${{ matrix.target }} ${{ matrix.extra-args }} -- -D warnings
+
+      - name: Cargo test
+        uses: clechasseur/rs-cargo@v2
+        if: startsWith(matrix.os, 'ubuntu')
+        with:
+          use-cross: ${{ matrix.cross }}
+          command: test
+          args: --all --target ${{ matrix.target }} ${{ matrix.extra-args }}
+        env:
+          CROSS_CONTAINER_OPTS: "--network host"
+
+      - name: Cargo test (no docker test)
+        uses: clechasseur/rs-cargo@v2
+        if: ${{ !startsWith(matrix.os, 'ubuntu') }}
+        with:
+          use-cross: ${{ matrix.cross }}
+          command: test
+          args: --all --target ${{ matrix.target }} ${{ matrix.extra-args }}
+        env:
+          CLASH_RS_CI: "true"
+
+      - name: Cargo build
+        uses: clechasseur/rs-cargo@v2
+        with:
+          use-cross: ${{ matrix.cross }}
+          command: build
+          args: --release --target ${{ matrix.target }} ${{ matrix.extra-args }}
+
+      - name: Rename binary
+        run: mv target/${{ matrix.target }}/release/${{ env.PACKAGE }}${{ matrix.postfix }} ${{ env.PACKAGE }}-${{ matrix.release-name }}${{ matrix.postfix }}
+
+      - name: Upload binaries
+        uses: actions/upload-artifact@v4
+        with:
+          name: binary-${{ matrix.target }}-${{ env.PACKAGE }}
+          path: ${{ env.PACKAGE }}-${{ matrix.release-name }}${{ matrix.postfix }}
+
+  release:
+    name: Release
+
+    needs: [compile]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Merge binaries
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: binaries
+          pattern: binary-*
+
+      - name: Download binaries
+        uses: actions/download-artifact@v4
+        with:
+          name: binaries
+          path: ./packages
+
+      - name: Github stable release
+        uses: "mathieucarbou/marvinpinto-action-automatic-releases@master"
+        if: startsWith(github.ref, 'refs/tags/v')
+        with:
+          repo_token: "${{ secrets.GITHUB_TOKEN }}"
+          prerelease: false
+          files: |
+            packages/*
+            LICENSE
+
+      - name: Github nightly release
+        uses: "mathieucarbou/marvinpinto-action-automatic-releases@master"
+        if: startsWith(github.ref, 'refs/heads/master')
```

Comment:
this might create a lot of releases as we constantly have auto package updates.

can we filter out by for example, PR tags/title/comments etc?

Reply:
> this might create a lot of releases 

Actually no. This release action always keep just one pre-release tag `latest`, it auto delete all previous pre-releases

Reply:
i see, that's good then